addons/binding/org.openhab.binding.unifi/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,137 @@
+				<label>Refresh Interval</label>
+				<description>The refresh interval in seconds to poll the UniFi controller</description>
+				<default>10</default>
+			</parameter>
+			<parameter name="considerHome" type="integer" required="false">
+				<label>Consider Home Interval</label>
+				<description>The interval in seconds used to consider a client as home</description>
+				<default>180</default>
+			</parameter>
+		</config-description>
+
+	</bridge-type>
+
+	<thing-type id="client">
+
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="controller" />
+		</supported-bridge-type-refs>
+		<label>UniFi Wireless Client</label>
+		<description>A wireless client connected to a UniFi wireless network</description>
+
+		<channels>
+			<channel id="online" typeId="online"/>
+			<channel id="site" typeId="site" />
+			<channel id="ap" typeId="ap" />
+			<channel id="essid" typeId="essid" />
+			<channel id="rssi" typeId="rssi" />
+			<channel id="uptime" typeId="uptime" />
+			<channel id="lastSeen" typeId="lastSeen" />
+		</channels>
+
+		<config-description>
+			<parameter name="mac" type="text" required="true">
+				<label>MAC Address</label>
+				<description>The MAC address of the client</description>
+			</parameter>
+			<parameter name="site" type="text" required="false">
+				<label>Site</label>
+				<description>The UniFi site where the client should be found (optional)</description>
+			</parameter>
+			<parameter name="contactType" type="text" required="false">
+				<label>Contact Type</label>
+				<description>The contact type for the online channel</description>
+				<options>
+					<option value="NO">Normally Open</option>
+					<option value="NC">Normally Closed</option>
+				</options>
+				<limitToOptions>true</limitToOptions>
+				<default>NO</default>
+			</parameter>
+		</config-description>
+		
+	</thing-type>
+	
+	<channel-type id="online">
+		<item-type>Contact</item-type>
+		<label>Online</label>
+		<description>Online status of the wireless client</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	
+	<channel-type id="ap">
+		<item-type>String</item-type>
+		<label>Access Point</label>
+		<description>Access Point the wireless client is connected to</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	
+	<channel-type id="essid">
+		<item-type>String</item-type>
+		<label>Wireless Network</label>
+		<description>Wireless Network (ESSID) the wireless client is connected to</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	
+	<channel-type id="rssi">
+		<item-type>Number</item-type>
+		<label>Received Signal Strength Indicator</label>
+		<description>Received Signal Strength Indicator (RSSI) of the wireless client</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	
+	<channel-type id="site">
+		<item-type>String</item-type>
+		<label>Site Name</label>
+		<description>UniFi Site the client is associated with</description>
```

Comment:
What is the Site and can that change dynamically? If not, that should rather be an (informative) Thing property.

Reply:
@kaikreuzer I'll chime in on this one since I was the one who asked for this feature. ;-)

A Unifi controller can manage multiple sites (a site is a place where you have Unifi switches, APs, etc. deployed). In the case where you have a controller managing more than one site (e.g. Home A and Home B), and you also have OH instances in each site, you need a way to tell the binding the site for which you want to know if a client (e.g. phone) is present. Without the ability to specify the site on the client thing, the binding would report the client present in Home A when the device is physically located in Home B (and vice versa).

Additional discussion here.
https://community.openhab.org/t/ubiquiti-unifi-binding-feature-discussion/14520/63

Reply:
Following up on this, yes the site can change dynamically. I have one instance of the UniFi controller that manages my home and office. As I move from my home to my office, my phone's site changes from "Home" to "Work". If my phone goes offline and the last site was "Work", I'm on my way home.

Reply:
ok, thanks for the explanation!